### PR TITLE
TIP-863: Move catalog bundle content

### PR DESCRIPTION
**Pull request by jjanvier**
This PR moves all classes of Catalog bundle into Enrichment.
PimCatalogBundle is deactivated, but the folder still exists. There are still classes related to Pim Version inside. We need to know where to put them.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Y
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php [hunk 1]
```diff
@@ -34,13 +34,13 @@ class ConfigureHistoryGridListener
     protected $requestParams;
 
     /**
-     * @var FQCNResolver
+     * @var \Akeneo\Pim\Enrichment\Bundle\Resolver\FQCNResolver
```
Comment (arnolanglade) [inline src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php]: to revert :trool:

Comment (arnolanglade) [inline src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php]: to revert :trollface:

File: src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php [hunk 2]
```diff
@@ -34,13 +34,13 @@ class ConfigureHistoryGridListener
...
      */
     protected $FQCNResolver;
 
     /**
-     * @param RequestParameters $requestParams
-     * @param FQCNResolver      $FQCNResolver
+     * @param RequestParameters                                   $requestParams
+     * @param \Akeneo\Pim\Enrichment\Bundle\Resolver\FQCNResolver $FQCNResolver
```
Comment (arnolanglade) [inline src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php]: idem :trolleybus:

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php [hunk 3]
```diff
@@ -20,7 +20,7 @@ class VersioningController
     /** @var VersionRepositoryInterface */
     protected $versionRepository;
 
-    /** @var FQCNResolver */
+    /** @var \Akeneo\Pim\Enrichment\Bundle\Resolver\FQCNResolver */
```
Comment (arnolanglade) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php]: to revert :trollface:

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php [hunk 4]
```diff
@@ -30,10 +30,10 @@ class VersioningController
     protected $userContext;
 
     /**
-     * @param VersionRepositoryInterface $versionRepository
-     * @param FQCNResolver               $FQCNResolver
-     * @param NormalizerInterface        $normalizer
-     * @param UserContext                $userContext
+     * @param VersionRepositoryInterface                          $versionRepository
+     * @param \Akeneo\Pim\Enrichment\Bundle\Resolver\FQCNResolver $FQCNResolver
```
Comment (arnolanglade) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php]: to revert :trolleybus:

File: src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php (hunk 1 above)
Comment (jjanvier) [reply src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php]: fixed

Comment (jjanvier) [reply src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php]: fixed

File: src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php (hunk 2 above)
Comment (jjanvier) [reply src/Pim/Bundle/DataGridBundle/EventListener/ConfigureHistoryGridListener.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php (hunk 3 above)
Comment (jjanvier) [reply src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php (hunk 4 above)
Comment (jjanvier) [reply src/Pim/Bundle/EnrichBundle/Controller/Rest/VersioningController.php]: fixed

**Pull request merged**